src/js/pouch-express.js

```diff
@@ -0,0 +1,288 @@
+    });
+
+    var sequence = fluid.promise.sequence(promises);
+    sequence.then(function () {
+        that.events.onStarted.fire();
+    });
+
+    return sequence;
+};
+
+/**
+ *
+ * Initialize a single database instance.
+ *
+ * @param that - The component itself
+ * @param dbKey {String} - The database name.
+ * @param dbDef {Object} - Our convention for representing multiple databases.  See the docs for examples.
+ * @returns {Promise} A promise that will be resolved with the database is initialized.
+ */
+gpii.pouch.express.initDb = function (that, dbKey, dbDef) {
+    var expandedDef = gpii.pouch.express.expandDbDef(dbDef);
+    var initPromise = fluid.promise();
+    var dbOptions = expandedDef.dbOptions ? fluid.merge(that.options.dbOptions, expandedDef.dbOptions) : fluid.copy(that.options.dbOptions);
+    dbOptions.name = dbKey;
+    var dbComponentOptions = {
+        type: "fluid.component",
+        gradeNames: that.options.pouchGradeNames,
+        dbOptions: dbOptions,
+        baseDir: that.options.baseDir,
+        listeners: {
+            "onDataLoaded.resolvePromise": {
+                func: initPromise.resolve
+            }
+        }
+    };
+    if (expandedDef.data) {
+        dbComponentOptions.dbPaths = expandedDef.data;
+    }
+
+    var dbComponent = fluid.construct("gpii_pouch_" + that.id + "_" + dbKey, dbComponentOptions);
+    that.databaseInstances[dbKey] = dbComponent;
+
+    return initPromise;
+};
+
+/**
+ *
+ * Pass along any requests to our instance of express-pouchdb.
+ *
+ * @param that - The component itself.
+ * @param req {Object} - The Express request object.
+ * @param res {Object} - The Express response object.
+ * @param next {Function} - The next piece of middleware in the chain.
+ *
+ */
+gpii.pouch.express.middleware = function (that, req, res, next) {
+    // fluid.log("express pouchdb instance '", that.id, "' responding...")
+    that.expressPouchdb(req, res, next);
+};
+
+/**
+ *
+ * Clean up any instance of `gpii.pouchdb` that we're aware of.  Then, to get rid of databases created by
+ * express-pouchdb, complete remove all data in `options.baseDir`.
+ *
+ * @param that - The component itself.
+ * @returns {Promise} - A promise that will be resolved when cleanup is complete.
+ *
+ */
+gpii.pouch.express.cleanup = function (that) {
+    // fluid.log("express pouchdb instance '", that.id, "' cleaning up...");
+    var promises = [];
+
+    fluid.each(that.databaseInstances, function (databaseInstance) {
+        promises.push(databaseInstance.destroyPouch());
+    });
+
+    var sequence = fluid.promise.sequence(promises);
+    sequence.then(function () {
+        // We cannot simply call db.destroy() on all databases, because express-pouchdb creates a few databases on its own
+        // and gives us no way to clean those up.
+        //
+        // The surest approach is to completely remove the underlying directory.  However, express-pouchdb will have one
```

Comment:
This is what I mean: https://pouchdb.com/adapters.html

We would use the default disk-based adapter as our starting point.